regen/warnings.pl

```diff
@@ -392,6 +392,11 @@ sub mkOct
 
 #define DUP_WARNINGS(p) Perl_dup_warnings(aTHX_ p)
 
+#define free_and_set_cop_warnings(cmp,w) STMT_START { \
+  if (!specialWARN(cmp.cop_warnings)) PerlMemShared_free(cmp.cop_warnings); \
+  cmp.cop_warnings = (STRLEN*) w; \
```

Comment:
I think the cast should only be in the one place that needs it - the scope.c call.

Reply:
I can move it back to scope.c and utf8.c

Reply:
I've moved the cast outside the macro